container/multi-value-slice/multi_value_slice.go

```diff
@@ -0,0 +1,380 @@
+package multi_value_slice
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/prysmaticlabs/prysm/v4/container/multi-value-slice/interfaces"
+)
+
+// MultiValueSlice defines an abstraction over all concrete implementations of the generic Slice.
+type MultiValueSlice[O interfaces.Identifiable] interface {
+	Len(obj O) int
+}
+
+// Value defines a single value along with one or more IDs that share this value.
+type Value[V any] struct {
+	val V
+	ids []interfaces.Id
+}
+
+// MultiValue defines a collection of Value items.
```

Comment:
Im not as a clear in the difference between this MultiValue type and the Slice type when it comes to naming. Maybe we need a better way to differentiate them. Perhaps, `SharedValueManager` instead of `Slice` ?

Reply:
I renamed `MultiValue` to `MultiValueItem`. I wouldn't call this a `Manager` because after all it's just a container.